### PR TITLE
feat(azure): add configuration for Azure OpenAI Service

### PR DESCRIPTION
**Pull request by unique-dominik**
Add an `azure` configuration option to support Azure OpenAI API calls in addition to the original OpenAI API calls.

To keep the diff clean, the commit does not involve a compiled version of the package in the `/dist` folder.

Reviewed-by: @konsti 
Refs: https://github.com/1openwindow/azure-openai-node

### CODE REVIEW

File: common.ts [hunk 1]
```diff
@@ -148,3 +149,29 @@ export const createRequestFunction = function (axiosArgs: RequestArgs, globalAxi
...
+export const createPrompt = function (systemMessage: string, messages: { sender: string, text: string }[]): string {
+    let prompt = systemMessage;
+    for (const message of messages) {
+        prompt += `\n<|im_start|>${message.sender}\n${message.text}\n<|im_end|>`;
+    }
+    prompt += "\n<|im_start|>assistant\n";
+    return prompt;
+}
+
+export const messageToAzurePrompt = function (messages: Array<ChatCompletionRequestMessage>): string {
```
Comment (unique-dominik) [inline common.ts:166]: @konsti I am not yet sure what these are for I do not see them in the readme.

File: README.md [hunk 2]
```diff
@@ -1,89 +1,70 @@
...
+> **Note**
+> It is also an inofficial fork of [1openwindow/azure-openai-node](https://github.com/1openwindow/azure-openai-node) which we could not use due to due diligence reasons 🥇 
+
+This library allows you to use Azure OpenAI's API without making any changes to your existing OpenAI code. You can simply add Azure information to your configuration and start using the Azure OpenAI model.
 
 ## Installation
 
 ```bash
-npm install openai
+$ npm install @unique/openai
```
Comment (konsti) [inline README.md:13]: We need to see if that namespace is available before merging.

File: README.md [hunk 3]
```diff
@@ -1,89 +1,70 @@
...
 ```
 
 ## Usage
 
-The library needs to be configured with your account's secret key, which is available on the [website](https://platform.openai.com/account/api-keys). We recommend setting it as an environment variable. Here's an example of initializing the library with the API key loaded from an environment variable and creating a completion:
+The library needs to be configured with your Azure OpenAI's key, endpoint and deploymentId. you can obtain these credentials from [Azure Portal](https://portal.azure.com).
```
Comment (konsti) [inline README.md:18]:
```suggestion
The library must be configured with your Azure OpenAI's `key`, `endpoint`, and `deploymentId`. You can obtain these credentials from [Azure Portal](https://portal.azure.com).
```

File: README.md [hunk 4]
```diff
@@ -1,89 +1,70 @@
...
 
-```javascript
-const { Configuration, OpenAIApi } = require("openai");
+```ts
+import { Configuration, OpenAIApi } from '@unique/openai';
```
Comment (konsti) [inline README.md:21]: see above: might be another namespace.

File: README.md [hunk 5]
```diff
@@ -1,89 +1,70 @@
...
-const openai = new OpenAIApi(configuration);
+
+const openai = new OpenAIApi(
+  new Configuration({
+    apiKey: this.apiKey,
+    // add azure info into configuration
+    azure: {
+      apiKey: <key1>
+      endpoint: <endpoint>,
+    // deploymentName is optional, if you donot set it, you need to set it in the request parameter
```
Comment (konsti) [inline README.md:34]:
```suggestion
    // The `deploymentName` parameter is optional; if you do not set it, you need to put it in the request parameter
```

File: api.ts [hunk 6]
```diff
@@ -1938,7 +1944,14 @@ export const OpenAIApiAxiosParamCreator = function (configuration?: Configuratio
         createChatCompletion: async (createChatCompletionRequest: CreateChatCompletionRequest, options: AxiosRequestConfig = {}): Promise<RequestArgs> => {
             // verify required parameter 'createChatCompletionRequest' is not null or undefined
             assertParamExists('createChatCompletion', 'createChatCompletionRequest', createChatCompletionRequest)
-            const localVarPath = `/chat/completions`;
+
+            let localVarPath = `/chat/completions`;
+            if (configuration.azure) {
+                let deploymentName = configuration.azure.deploymentName ? configuration.azure.deploymentName : createChatCompletionRequest.model;
+                let apiVersion = configuration.azure.apiVersion ? configuration.azure.apiVersion : '2023-03-15-preview';
+                localVarPath = `/openai/deployments/${deploymentName}/chat/completions?api-version=${configuration.azure.apiVersion}`;
```
Comment (konsti) [inline api.ts:1952]:
```suggestion
                const deploymentName = configuration.azure.deploymentName ? configuration.azure.deploymentName : createChatCompletionRequest.model;
                const apiVersion = configuration.azure.apiVersion ? configuration.azure.apiVersion : '2023-03-15-preview';
                localVarPath = `/openai/deployments/${deploymentName}/chat/completions?api-version=${configuration.azure.apiVersion}`;
```

File: api.ts [hunk 7]
```diff
@@ -2083,7 +2101,12 @@ export const OpenAIApiAxiosParamCreator = function (configuration?: Configuratio
         createEmbedding: async (createEmbeddingRequest: CreateEmbeddingRequest, options: AxiosRequestConfig = {}): Promise<RequestArgs> => {
             // verify required parameter 'createEmbeddingRequest' is not null or undefined
             assertParamExists('createEmbedding', 'createEmbeddingRequest', createEmbeddingRequest)
-            const localVarPath = `/embeddings`;
+            let localVarPath = `/embeddings`;
+            if (configuration.azure) {
+                let deploymentName = configuration.azure.deploymentName ? configuration.azure.deploymentName : createEmbeddingRequest.model;
+                let apiVersion = configuration.azure.apiVersion ? configuration.azure.apiVersion : '2023-03-15-preview';
```
Comment (konsti) [inline api.ts:2107]:
```suggestion
                const deploymentName = configuration.azure.deploymentName ? configuration.azure.deploymentName : createEmbeddingRequest.model;
                const apiVersion = configuration.azure.apiVersion ? configuration.azure.apiVersion : '2023-03-15-preview';
```

File: api.ts [hunk 8]
```diff
@@ -13,15 +13,15 @@
...
-import type { Configuration } from './configuration';
-import type { AxiosPromise, AxiosInstance, AxiosRequestConfig } from 'axios';
+import type { AxiosInstance, AxiosPromise, AxiosRequestConfig } from 'axios';
 import globalAxios from 'axios';
+import type { Configuration } from './configuration';
 // Some imports not used depending on template conditions
 // @ts-ignore
-import { DUMMY_BASE_URL, assertParamExists, setApiKeyToObject, setBasicAuthToObject, setBearerAuthToObject, setOAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction } from './common';
 import type { RequestArgs } from './base';
+import { DUMMY_BASE_URL, assertParamExists, createRequestFunction, serializeDataIfNeeded, setSearchParams, toPathString } from './common';
```
Comment (konsti) [inline api.ts:22]: Why where these imports removed? If you read the comment in line 19 it seams they are needed.

File: common.ts (hunk 1 above)
Comment (konsti) [reply common.ts:166]: I wonder why those two functions were added. I think they are bloating the API un-necessarily and can be removed.

File: configuration.ts [hunk 9]
```diff
@@ -93,12 +105,13 @@ export class Configuration {
...
         this.baseOptions = param.baseOptions;
         this.formDataCtor = param.formDataCtor;
+        this.azure = param.azure;
 
         if (!this.baseOptions) {
             this.baseOptions = {};
         }
         this.baseOptions.headers = {
-            'User-Agent': `OpenAI/NodeJS/${packageJson.version}`,
+            'User-Agent': `AzureQOpenAI/NodeJS/${packageJson.version}`,
```
Comment (konsti) [inline configuration.ts:114]:
```suggestion
            'User-Agent': `AzureOpenAI/NodeJS/${packageJson.version}`,
```

File: api.ts (hunk 8 above)
Comment (unique-dominik) [reply api.ts:22]: From `Organise imports on save`. Will add them back.

**Pull request merged**